### PR TITLE
Don't hit the database for retrieving packages #2527

### PR DESCRIPTION
**Pull request by maartenba**
Don't merge it just yet.

* Fixed OData feed (FindPackagesById())
* Package details page pending.

### CODE REVIEW

File: src/NuGetGallery/DataServices/V2Feed.svc.cs [hunk 1]
```diff
@@ -156,9 +156,14 @@ private IQueryable<V2FeedPackage> SearchV2FeedCore(string searchTerm, string tar
         [WebGet]
         public IQueryable<V2FeedPackage> FindPackagesById(string id)
         {
-            return PackageRepository.GetAll().Include(p => p.PackageRegistration)
-                .Where(p => p.PackageRegistration.Id.Equals(id, StringComparison.OrdinalIgnoreCase))
+            var packages = PackageRepository.GetAll().Include(p => p.PackageRegistration)
```
Comment (yishaigalatzer) [inline src/NuGetGallery/DataServices/V2Feed.svc.cs]: lets follow the coding standards and put each linq statement on a separate line

File: src/NuGetGallery/DataServices/V2Feed.svc.cs [hunk 2]
```diff
@@ -156,9 +156,14 @@ private IQueryable<V2FeedPackage> SearchV2FeedCore(string searchTerm, string tar
...
+                .Where(p => p.PackageRegistration.Id.Equals(id, StringComparison.OrdinalIgnoreCase));
+
+            var query = SearchAdaptor.FindByIdCore(SearchService, HttpContext.Request, packages, id, curatedFeed: null)
+                .Result
```
Comment (yishaigalatzer) [inline src/NuGetGallery/DataServices/V2Feed.svc.cs]: .Result can start choking the asp.net threads, should we bump them up as part of this change?

**Comment by yishaigalatzer** [comment]
:shipit: when feedback addressed, and more eyes have taken a look

File: src/NuGetGallery/DataServices/V2Feed.svc.cs (hunk 2 above)
Comment (maartenba) [inline src/NuGetGallery/DataServices/V2Feed.svc.cs]: Yes. Will be in a separate commit. When moving to Web API OData we can do proper async as well.

**Comment by maartenba** [comment]
@bhuvak @johnataylor @xavierdecoster Can one of you have a look at this one? Would like to :shipit:  soon.

**Comment by xavierdecoster** [comment]
:shipit:
